src/inc/regdisp.h

```diff
@@ -8,7 +8,7 @@
 
 #ifdef DEBUG_REGDISPLAY
 class Thread;
-
+struct REGDISPLAY;
```

Comment:
I agree, I doesn't matter whether we move the prototype after the definitions or keep it here and use the forward declaration.